src/events.rs

```diff
@@ -339,35 +339,187 @@ pub enum EventsDecodingError {
     InvalidCompactType(String),
 }
 
-// #[cfg(test)]
-// mod tests {
-//     use super::*;
-//     use std::convert::TryFrom;
-//
-//     type DefaultConfig = crate::NodeTemplateRuntime;
-//
-//     #[test]
-//     fn test_decode_option() {
-//         let decoder = EventsDecoder::<DefaultConfig>::new(
-//             Metadata::default(),
-//         );
-//
-//         let value = Some(0u8);
-//         let input = value.encode();
-//         let mut output = Vec::<u8>::new();
-//         let mut errors = Vec::<RuntimeError>::new();
-//
-//         decoder
-//             .decode_raw_bytes(
-//                 &[EventArg::Option(Box::new(EventArg::Primitive(
-//                     "u8".to_string(),
-//                 )))],
-//                 &mut &input[..],
-//                 &mut output,
-//                 &mut errors,
-//             )
-//             .unwrap();
-//
-//         assert_eq!(output, vec![1, 0]);
-//     }
-// }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        Config,
+        DefaultConfig,
+        Phase,
+    };
+    use frame_metadata::{
+        v14::{
+            ExtrinsicMetadata,
+            PalletEventMetadata,
+            PalletMetadata,
+            RuntimeMetadataLastVersion,
+        },
+        RuntimeMetadataPrefixed,
+    };
+    use scale_info::{
+        meta_type,
+        TypeInfo,
+    };
+    use std::convert::TryFrom;
```

Comment:
we are on `edition 2021` right?

```suggestion
```

Reply:
Yes we are, sorry I missed this comment before merging :grimacing: